pkg/deploy/api/v1beta3/types.go

```diff
@@ -67,9 +67,15 @@ type CustomDeploymentStrategyParams struct {
 // RecreateDeploymentStrategyParams are the input to the Recreate deployment
 // strategy.
 type RecreateDeploymentStrategyParams struct {
+	// TimeoutSeconds is the time to wait for updates before giving up. If the
+	// value is nil, a default will be used.
+	TimeoutSeconds *int64 `json:"timeoutSeconds,omitempty" description:"the time to wait for updates before giving up"`
```

Comment:
That change isn't backwards compatible - it would have to be zero or
negative one.  Recreate already had an implicit timeout of 120s.